x-pack/plugins/observability_ai_assistant/public/functions/index.ts

```diff
@@ -5,31 +5,68 @@
  * 2.0.
  */
 
+import dedent from 'dedent';
 import type { RegisterContextDefinition, RegisterFunctionDefinition } from '../../common/types';
 import type { ObservabilityAIAssistantPluginStartDependencies } from '../types';
 import type { ObservabilityAIAssistantService } from '../types';
 import { registerElasticsearchFunction } from './elasticsearch';
 import { registerRecallFunction } from './recall';
 import { registerSummarisationFunction } from './summarise';
 
-export function registerFunctions({
+export async function registerFunctions({
   registerFunction,
   registerContext,
   service,
   pluginsStart,
+  signal,
 }: {
   registerFunction: RegisterFunctionDefinition;
   registerContext: RegisterContextDefinition;
   service: ObservabilityAIAssistantService;
   pluginsStart: ObservabilityAIAssistantPluginStartDependencies;
+  signal: AbortSignal;
 }) {
-  registerContext({
-    name: 'core',
-    description:
-      'Core functions, like calling Elasticsearch APIs, storing embeddables for instructions or creating base visualisations.',
-  });
+  return service
+    .callApi('GET /internal/observability_ai_assistant/functions/kb_status', {
+      signal,
+    })
+    .then((response) => {
+      const isReady = response.ready;
 
-  registerElasticsearchFunction({ service, registerFunction });
-  registerSummarisationFunction({ service, registerFunction });
-  registerRecallFunction({ service, registerFunction });
+      let description = `You have the ability to call Elasticsearch APIs or create visualisations using Lens.`;
+
+      if (isReady) {
+        description += `You can use the "summarise" functions to store new information you have learned in a knowledge database. Once you have established that you did not know the answer to a question, and the user gave you this information, it's important that you create a summarisation of what you have learned and store it in the knowledge database. When you create this summarisation, make sure you craft it in a way that can be recalled with a semantic search later.
+
+        Additionally, you can use the "recall" function to retrieve relevant information from the knowledge database. Using the "recall" function will allow you to create a much more tailored and improved user experience. Assume the user has told you before.
+      
+        For every question that the user asks, you, as the assistant, must query the knowledge base, no matter how confident you are in your response.
```

Comment:
Will this mean the user will, for every prompt, see a "the assistant has suggested 'recall'" and "you executed function 'recall'"?

Reply:
I'm not sure if I follow, this is part of the system prompt (that the user doesn't see).